### ISSUE TITLE
Add a config file for consistent formatting in vscode

### ISSUE THREAD

**Issue by jsergiu** [Freeyayo/ifoo#6]
Prettier or editorconfig should do. I don't have any strong opinions on how to format it but I think having a config file for this would help keeping the code consistent.

**Comment by Freeyayo** [comment]
I'm considering adding ESlint to our project, any better option?

**Comment by Freeyayo** [comment]
I prefer to use ESlint

**Comment by SapeleD3** [comment]
i intend adding Eslint rules and husky for git pre commit .. i just need a list of absolute must 'off' eslint rules for contributors

**Comment by Freeyayo** [comment]
I am thinking whether `console.log` a good error reminder ..or not. 

`console` is regarded as `side-effect` in many situations. But `var` shouldn't exist anyway

**Comment by SapeleD3** [comment]
You can use the debug package to create a log util that way you can log errors on development but after build on productions the would not be transpiled

**Comment by SapeleD3** [comment]
![logger](https://user-images.githubusercontent.com/46188071/86248138-5ea39c00-bba5-11ea-826f-2f0bf10bab9b.png)

**Comment by Freeyayo** [comment]
It's nice to debug when we are on deployment. We may add it to our project in future. Thx

**Comment by Freeyayo** [comment]
in commit `f1adf62` I killed all the errors reported from eslint, but still keep "any" warnings. 

we should keep our code as be typed as possible.

many types is difficult to be implemented, but they are worth it.

**Comment by Freeyayo** [comment]
I just ignore all `:any` warnings in our project

**Comment by Freeyayo** [comment]
> i intend adding Eslint rules and husky for git pre commit .. i just need a list of absolute must 'off' eslint rules for contributors

@SapeleD3 
This week is super busy and happening to be involved in a project that using husky and pre commit checking. It’s useful when we do a big project but I think it’s a little bit over engineered if we add pre commit checking or formatting to our library. 
How do you think about it?

**Comment by SapeleD3** [comment]
True.  Spurv has a knit pick team so there is no need for pre commit hooks
